File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int r=0;
        int sum = 0;
        int ans = INT_MIN;
        while(r<nums.size())
        {
            sum += nums[r];
            ans = max(ans,sum);
            if(sum < 0)
            {
                sum = 0;
            }
            
            r++;
            
            
            
        }
        return ans;


    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        dp = {}
        def f(i):
            if i >= len(nums):
                return 0
            if i in dp:
                return dp[i]
            
            l = f(i+2) + nums[i]
            r = f(i+1)
            dp[i] = max(l,r)
            return max(l,r)

        return f(0)